src/app/modules/products/pages/product/product.page.html

```diff
@@ -1,8 +1,36 @@
 <div class="product-page-container">
     <app-grid-list [itemColumnSpan]=2 [gridGap]=4>
-        <app-featured-product-card *ngFor="let product of featuredProducts$ | async" [product]="product"></app-featured-product-card>
+        <ng-container *ngIf="featuredProducts$ | async as featuredProducts; else featuredProductsLoading">
+            <app-featured-product-card *ngFor="let product of featuredProducts" [product]="product"></app-featured-product-card>
+        </ng-container>
```

Comment:
this is a good start.

the ngx-skeleton-loader and its ngIf should be in the component that it replaces. Here, in featured-product-card, we should check whether the product is available and if not, display the loader. Same for product-card and product-detail

The page components should not know whether the result will be a loader or an actual component, they should only be concerned about what will take up that specific space in the layout